src/ui/components/forms/VPNTextArea.qml

```diff
@@ -76,6 +78,28 @@ Item {
                 id: textAreaStates
                 itemToTarget: textArea
             }
+
+            function removeCharsInRange(string, startIndex, endIndex) {
+                return string.slice(0, startIndex) + string.slice(endIndex);
+            }
+
+            function handleOnTextChanged(text) {
```

Comment:
what is 'text' here? It's the new added text or the entire one in the textarea?

Reply:
It’s the entire text in the textarea after the text changed.